trackers/dataset/base.py

```diff
@@ -0,0 +1,87 @@
+import abc
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import supervision as sv
+
+
+# --- Base Dataset ---
+class Dataset(abc.ABC):
+    """Abstract base class for datasets used in tracking evaluation."""
+
+    @abc.abstractmethod
+    def load_ground_truth(self, sequence_name: str) -> Optional[sv.Detections]:
```

Comment:
It seems like `Union[sv.Detections, None]` would be a more suitable return type here. @onuralpszr WDYT?

Reply:
Optional[...] is a shorthand notation for Union[..., None] so it is the same thing :)